zerver/views/auth.py

```diff
@@ -62,11 +62,21 @@
 ExtraContext = Optional[Dict[str, Any]]
 
 def get_safe_redirect_to(url: str, redirect_host: str) -> str:
-    is_url_safe = is_safe_url(url=url, allowed_hosts=None)
+    # allowed_hosts given to is_safe_url are supposed to be without the scheme,
+    # e.g. example.com is valid, but https://example.com is not. We want to allow
+    # passing redirect_host with the scheme here, so we'll handle those cases by
+    # extracting the netloc.
+    parsed_host = urllib.parse.urlparse(redirect_host)
+    if parsed_host.scheme:
+        redirect_host = parsed_host.netloc
```

Comment:
I agree with Tim that I don’t see a clear use case for this commit, but if we’re going to do something like this, we should be unambiguous about whether the function expects a host or a URL, rather than trying to support both.

Reply:
That makes sense I think, if we decide we want this, I think expecting URL would be the right format, because that's how this is currently being called, and it would be a big hassle for callers to be expected to do this parsing.